packages/office-ui-fabric-react/src/utilities/positioning/positioning.ts

```diff
@@ -335,12 +334,31 @@ function _adjustFitWithinBounds(
       const flippedElementEstimate = _flipAlignmentEdge(elementEstimate, target, gap, coverTarget);
       if (_isRectangleWithinBounds(flippedElementEstimate.elementRectangle, bounding)) {
         return flippedElementEstimate;
+      } else {
+        // If the flipped elements edges are still out of bounds, try nudging it.
+        elementEstimate = _alignOutOfBoundsEdges(
```

Comment:
It was a choice between having it clipped because it was out of bounds or covering the target - in our actual scenarios this isnt a problem since we only need it nudged in one direction and that's usually along the aligment edge. Open to any recommendations you may have to make this better.